ArduCopter/mode_guided.cpp

```diff
@@ -768,6 +780,159 @@ bool ModeGuided::limit_check()
     return false;
 }
 
+//loiter turns(circle) mode
+
+void ModeGuided::do_circle(const mavlink_command_long_t &packet)
```

Comment:
We probably don't want to be passing in a `mavlink_command_long_t` here.  Note that the other guided-submodes don't pass in their packets either.

We'll need to break out the relevant parameters in the GCS_MAVLink library and feed them in here.  That way if someone wants to use guided mode from a script they can easily do so, too.